[Bug]: Flink Ingestion jobs will occasionally delete snapshot files causing Optimize and Flink Job to fail

### What happened?

When the Flink Ingestion job failsover several times, restarting the job will occasionally result in a java.io.FileNotFoundException: File does not exist exception for the snap-xxx.avro file.

### Affects Versions

0.3.x

### What engines are you seeing the problem on?

Flink

### How to reproduce

After starting a Flink job and writing some data normally, it frequently triggers Failover for a period of time causing the job to fail. Restart the task you just started.

### Relevant log output

```shell
java.lang.reflect.InvocationTargetException at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at 
sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at 
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at 
java.lang.reflect.Method.invoke(Method.java:498) at 
com.netease.arctic.flink.interceptor.KerberosInvocationHandler.lambda$invoke$0(KerberosInvocationHandler.java:58) at 
com.netease.arctic.table.TableMetaStore.lambda$doAs$0(TableMetaStore.java:301) at 
java.security.AccessController.doPrivileged(Native Method) at javax.security.auth.Subject.doAs(Subject.java:360) at 
org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1734) at 
com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:299) at 
com.netease.arctic.io.ArcticHadoopFileIO.doAs(ArcticHadoopFileIO.java:177) at 
com.netease.arctic.flink.interceptor.KerberosInvocationHandler.invoke(KerberosInvocationHandler.java:55) at 
com.sun.proxy.$Proxy37.initializeState(Unknown Source) at 
org.apache.flink.streaming.runtime.tasks.OperatorChain.initializeStateAndOpenOperators(OperatorChain.java:427) at 
org.apache.flink.streaming.runtime.tasks.StreamTask.lambda$beforeInvoke$2(StreamTask.java:545) at 
org.apache.flink.streaming.runtime.tasks.StreamTaskActionExecutor$1.runThrowing(StreamTaskActionExecutor.java:50) at 
org.apache.flink.streaming.runtime.tasks.StreamTask.beforeInvoke(StreamTask.java:535) at 
org.apache.flink.streaming.runtime.tasks.StreamTask.invoke(StreamTask.java:575) at 
org.apache.flink.runtime.taskmanager.Task.doRun(Task.java:758) at org.apache.flink.runtime.taskmanager.Task.run(Task.java:573) 
at java.lang.Thread.run(Thread.java:745) Caused by: java.lang.reflect.InvocationTargetException at 
sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at 
sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at 
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at 
java.lang.reflect.Method.invoke(Method.java:498) at 
com.netease.arctic.flink.interceptor.KerberosInvocationHandler.lambda$invoke$0(KerberosInvocationHandler.java:58) at 
com.netease.arctic.table.TableMetaStore.lambda$doAs$0(TableMetaStore.java:301) at 
java.security.AccessController.doPrivileged(Native Method) at javax.security.auth.Subject.doAs(Subject.java:360) at 
org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1734) at 
com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:299) at 
com.netease.arctic.io.ArcticHadoopFileIO.doAs(ArcticHadoopFileIO.java:177) at 
com.netease.arctic.flink.interceptor.KerberosInvocationHandler.invoke(KerberosInvocationHandler.java:55) at 
com.sun.proxy.$Proxy37.initializeState(Unknown Source) ... 21 more Caused by: 
com.netease.arctic.shaded.org.apache.iceberg.exceptions.NotFoundException: Failed to open input stream for file: 
hdfs://eadhadoop/user/da_luna/hive_db/da_luna.db/arctic_wow_outlog/change/metadata/snap-4720345255491225694-1-
2100685a-488a-496f-88b5-58206170c497.avro at 
com.netease.arctic.shaded.org.apache.iceberg.hadoop.HadoopInputFile.newStream(HadoopInputFile.java:177) at 
com.netease.arctic.shaded.org.apache.iceberg.avro.AvroIterable.newFileReader(AvroIterable.java:101) at 
com.netease.arctic.shaded.org.apache.iceberg.avro.AvroIterable.iterator(AvroIterable.java:77) at 
com.netease.arctic.shaded.org.apache.iceberg.avro.AvroIterable.iterator(AvroIterable.java:37) at 
com.netease.arctic.shaded.org.apache.iceberg.relocated.com.google.common.collect.Iterables.addAll(Iterables.java:320) at 
com.netease.arctic.shaded.org.apache.iceberg.relocated.com.google.common.collect.Lists.newLinkedList(Lists.java:237) at 
com.netease.arctic.shaded.org.apache.iceberg.ManifestLists.read(ManifestLists.java:46) at 
com.netease.arctic.shaded.org.apache.iceberg.BaseSnapshot.cacheManifests(BaseSnapshot.java:137) at 
com.netease.arctic.shaded.org.apache.iceberg.BaseSnapshot.dataManifests(BaseSnapshot.java:159) at 
com.netease.arctic.shaded.org.apache.iceberg.MergingSnapshotProducer.apply(MergingSnapshotProducer.java:403) at 
com.netease.arctic.shaded.org.apache.iceberg.SnapshotProducer.apply(SnapshotProducer.java:164) at 
com.netease.arctic.shaded.org.apache.iceberg.SnapshotProducer.lambda$commit$2(SnapshotProducer.java:283) at 
com.netease.arctic.shaded.org.apache.iceberg.util.Tasks$Builder.runTaskWithRetry(Tasks.java:405) at 
com.netease.arctic.shaded.org.apache.iceberg.util.Tasks$Builder.runSingleThreaded(Tasks.java:214) at 
com.netease.arctic.shaded.org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:198) at 
com.netease.arctic.shaded.org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:190) at 
com.netease.arctic.shaded.org.apache.iceberg.SnapshotProducer.commit(SnapshotProducer.java:282) at 
com.netease.arctic.trace.TracedAppendFiles.commit(TracedAppendFiles.java:81) at 
com.netease.arctic.shaded.org.apache.iceberg.flink.sink.IcebergFilesCommitter.commitOperation(IcebergFilesCommitter.java:30
8) at 
com.netease.arctic.shaded.org.apache.iceberg.flink.sink.IcebergFilesCommitter.commitDeltaTxn(IcebergFilesCommitter.java:277) 
at 
com.netease.arctic.shaded.org.apache.iceberg.flink.sink.IcebergFilesCommitter.commitUpToCheckpoint(IcebergFilesCommitter.j
ava:219) at 
com.netease.arctic.shaded.org.apache.iceberg.flink.sink.IcebergFilesCommitter.initializeState(IcebergFilesCommitter.java:154) at 
org.apache.flink.streaming.api.operators.StreamOperatorStateHandler.initializeOperatorState(StreamOperatorStateHandler.java:
112) at org.apache.flink.streaming.api.operators.AbstractStreamOperator.initializeState(AbstractStreamOperator.java:290) ... 34 
more Caused by: java.io.FileNotFoundException: File does not exist: 
/user/da_luna/hive_db/da_luna.db/arctic_wow_outlog/change/metadata/snap-4720345255491225694-1-2100685a-488a-496f-
88b5-58206170c497.avro at org.apache.hadoop.hdfs.server.namenode.INodeFile.valueOf(INodeFile.java:86) at 
org.apache.hadoop.hdfs.server.namenode.INodeFile.valueOf(INodeFile.java:76) at 
org.apache.hadoop.hdfs.server.namenode.FSDirStatAndListingOp.getBlockLocations(FSDirStatAndListingOp.java:156) at 
org.apache.hadoop.hdfs.server.namenode.FSNamesystem.getBlockLocations(FSNamesystem.java:2099) at 
org.apache.hadoop.hdfs.server.namenode.NameNodeRpcServer.getBlockLocations(NameNodeRpcServer.java:780) at 
org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolServerSideTranslatorPB.getBlockLocations(ClientNamenodeProtoc
olServerSideTranslatorPB.java:458) at 
org.apache.hadoop.ipc.ProtobufRpcEngine2$Server$ProtoBufRpcInvoker.call(ProtobufRpcEngine2.java:604) at 
org.apache.hadoop.ipc.ProtobufRpcEngine2$Server$ProtoBufRpcInvoker.call(ProtobufRpcEngine2.java:572) at 
org.apache.hadoop.ipc.ProtobufRpcEngine2$Server$ProtoBufRpcInvoker.call(ProtobufRpcEngine2.java:556) at 
org.apache.hadoop.ipc.RPC$Server.call(RPC.java:1093) at org.apache.hadoop.ipc.Server$RpcCall.run(Server.java:1043) at 
org.apache.hadoop.ipc.Server$RpcCall.run(Server.java:971) at java.base/javax.security.auth.Subject.doAs(Subject.java:423) at 
org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1878) at 
org.apache.hadoop.ipc.Server$Handler.run(Server.java:2976) at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native 
Method) at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) at 
sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) at 
java.lang.reflect.Constructor.newInstance(Constructor.java:423) at 
org.apache.hadoop.ipc.RemoteException.instantiateException(RemoteException.java:106) at 
org.apache.hadoop.ipc.RemoteException.unwrapRemoteException(RemoteException.java:73) at 
org.apache.hadoop.hdfs.DFSClient.callGetBlockLocations(DFSClient.java:1237) at 
org.apache.hadoop.hdfs.DFSClient.getLocatedBlocks(DFSClient.java:1222) at 
org.apache.hadoop.hdfs.DFSClient.getLocatedBlocks(DFSClient.java:1210) at 
org.apache.hadoop.hdfs.DFSInputStream.fetchLocatedBlocksAndGetLastBlockLength(DFSInputStream.java:306) at 
org.apache.hadoop.hdfs.DFSInputStream.openInfo(DFSInputStream.java:272) at org.apache.hadoop.hdfs.DFSInputStream.<init>
(DFSInputStream.java:264) at org.apache.hadoop.hdfs.DFSClient.open(DFSClient.java:1535) at 
org.apache.hadoop.hdfs.DistributedFileSystem$3.doCall(DistributedFileSystem.java:304) at 
org.apache.hadoop.hdfs.DistributedFileSystem$3.doCall(DistributedFileSystem.java:299) at 
org.apache.hadoop.fs.FileSystemLinkResolver.resolve(FileSystemLinkResolver.java:81) at 
org.apache.hadoop.hdfs.DistributedFileSystem.open(DistributedFileSystem.java:312) at 
org.apache.hadoop.fs.FileSystem.open(FileSystem.java:768) at 
com.netease.arctic.shaded.org.apache.iceberg.hadoop.HadoopInputFile.newStream(HadoopInputFile.java:175) ... 57 more 
Caused by: org.apache.hadoop.ipc.RemoteException(java.io.FileNotFoundException): File does not exist: 
/user/da_luna/hive_db/da_luna.db/arctic_wow_outlog/change/metadata/snap-4720345255491225694-1-2100685a-488a-496f-
88b5-58206170c497.avro at org.apache.hadoop.hdfs.server.namenode.INodeFile.valueOf(INodeFile.java:86) at org.apache.hadoop.hdfs.server.namenode.INodeFile.valueOf(INodeFile.java:76) at 
org.apache.hadoop.hdfs.server.namenode.FSDirStatAndListingOp.getBlockLocations(FSDirStatAndListingOp.java:156) at 
org.apache.hadoop.hdfs.server.namenode.FSNamesystem.getBlockLocations(FSNamesystem.java:2099) at 
org.apache.hadoop.hdfs.server.namenode.NameNodeRpcServer.getBlockLocations(NameNodeRpcServer.java:780) at 
org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolServerSideTranslatorPB.getBlockLocations(ClientNamenodeProtoc
olServerSideTranslatorPB.java:458) at 
org.apache.hadoop.ipc.ProtobufRpcEngine2$Server$ProtoBufRpcInvoker.call(ProtobufRpcEngine2.java:604) at 
org.apache.hadoop.ipc.ProtobufRpcEngine2$Server$ProtoBufRpcInvoker.call(ProtobufRpcEngine2.java:572) at 
org.apache.hadoop.ipc.ProtobufRpcEngine2$Server$ProtoBufRpcInvoker.call(ProtobufRpcEngine2.java:556) at 
org.apache.hadoop.ipc.RPC$Server.call(RPC.java:1093) at org.apache.hadoop.ipc.Server$RpcCall.run(Server.java:1043) at 
org.apache.hadoop.ipc.Server$RpcCall.run(Server.java:971) at java.base/javax.security.auth.Subject.doAs(Subject.java:423) at 
org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1878) at 
org.apache.hadoop.ipc.Server$Handler.run(Server.java:2976) at org.apache.hadoop.ipc.Client.call(Client.java:1476) at 
org.apache.hadoop.ipc.Client.call(Client.java:1413) at 
org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:229) at 
com.sun.proxy.$Proxy27.getBlockLocations(Unknown Source) at 
org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.getBlockLocations(ClientNamenodeProtocolTranslato
rPB.java:255) at sun.reflect.GeneratedMethodAccessor12.invoke(Unknown Source) at 
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at 
java.lang.reflect.Method.invoke(Method.java:498) at 
org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:191) at 
org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:102) at 
com.sun.proxy.$Proxy28.getBlockLocations(Unknown Source) at 
org.apache.hadoop.hdfs.DFSClient.callGetBlockLocations(DFSClient.java:1235) ... 69 more


```


### Anything else

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

com.netease.arctic.shaded.org.apache.iceberg.exceptions.CommitFailedException: Failed to commit changes using rename: hdfs://easyops-cluster/user/arctic/hive_db/arctic.db/avro_lost_test/change/metadata/v801.metadata.json
	at com.netease.arctic.shaded.org.apache.iceberg.hadoop.HadoopTableOperations.renameToFinal(HadoopTableOperations.java:366) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.hadoop.HadoopTableOperations.commit(HadoopTableOperations.java:158) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.op.ArcticHadoopTableOperations.lambda$commit$1(ArcticHadoopTableOperations.java:45) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.table.TableMetaStore.lambda$doAs$0(TableMetaStore.java:315) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_152]
	at javax.security.auth.Subject.doAs(Subject.java:360) ~[?:1.8.0_152]
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1734) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:313) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.io.ArcticHadoopFileIO.doAs(ArcticHadoopFileIO.java:184) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.op.ArcticHadoopTableOperations.commit(ArcticHadoopTableOperations.java:44) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.BaseTransaction.lambda$commitSimpleTransaction$5(BaseTransaction.java:356) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.util.Tasks$Builder.runTaskWithRetry(Tasks.java:404) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.util.Tasks$Builder.runSingleThreaded(Tasks.java:214) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:198) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:190) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.BaseTransaction.commitSimpleTransaction(BaseTransaction.java:348) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.BaseTransaction.commitTransaction(BaseTransaction.java:238) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.op.ArcticUpdate.commit(ArcticUpdate.java:137) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.flink.sink.IcebergFilesCommitter.commitOperation(IcebergFilesCommitter.java:312) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.flink.sink.IcebergFilesCommitter.commitDeltaTxn(IcebergFilesCommitter.java:276) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.flink.sink.IcebergFilesCommitter.commitUpToCheckpoint(IcebergFilesCommitter.java:218) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.flink.sink.IcebergFilesCommitter.notifyCheckpointComplete(IcebergFilesCommitter.java:188) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at sun.reflect.GeneratedMethodAccessor54.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_152]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_152]
	at com.netease.arctic.flink.interceptor.KerberosInvocationHandler.lambda$invoke$0(KerberosInvocationHandler.java:58) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.table.TableMetaStore.lambda$doAs$0(TableMetaStore.java:315) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_152]
	at javax.security.auth.Subject.doAs(Subject.java:360) ~[?:1.8.0_152]
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1734) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:313) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.io.ArcticHadoopFileIO.doAs(ArcticHadoopFileIO.java:184) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.flink.interceptor.KerberosInvocationHandler.invoke(KerberosInvocationHandler.java:55) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.sun.proxy.$Proxy38.notifyCheckpointComplete(Unknown Source) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor54.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_152]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_152]
	at com.netease.arctic.flink.interceptor.KerberosInvocationHandler.lambda$invoke$0(KerberosInvocationHandler.java:58) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.table.TableMetaStore.lambda$doAs$0(TableMetaStore.java:315) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_152]
	at javax.security.auth.Subject.doAs(Subject.java:360) ~[?:1.8.0_152]
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1734) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:313) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.io.ArcticHadoopFileIO.doAs(ArcticHadoopFileIO.java:184) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.flink.interceptor.KerberosInvocationHandler.invoke(KerberosInvocationHandler.java:55) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.sun.proxy.$Proxy38.notifyCheckpointComplete(Unknown Source) ~[?:?]
	at org.apache.flink.streaming.runtime.tasks.StreamOperatorWrapper.notifyCheckpointComplete(StreamOperatorWrapper.java:99) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.streaming.runtime.tasks.SubtaskCheckpointCoordinatorImpl.notifyCheckpointComplete(SubtaskCheckpointCoordinatorImpl.java:319) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.notifyCheckpointComplete(StreamTask.java:1089) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.lambda$notifyCheckpointCompleteAsync$11(StreamTask.java:1054) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.lambda$notifyCheckpointOperation$13(StreamTask.java:1077) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.streaming.runtime.tasks.StreamTaskActionExecutor$1.runThrowing(StreamTaskActionExecutor.java:50) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.streaming.runtime.tasks.mailbox.Mail.run(Mail.java:90) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.processMail(MailboxProcessor.java:317) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.runMailboxLoop(MailboxProcessor.java:189) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.runMailboxLoop(StreamTask.java:619) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.invoke(StreamTask.java:583) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.runtime.taskmanager.Task.doRun(Task.java:758) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.runtime.taskmanager.Task.run(Task.java:573) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_152]
	Suppressed: java.io.UncheckedIOException: Failed to delete file: hdfs://easyops-cluster/user/arctic/hive_db/arctic.db/avro_lost_test/change/metadata/13322137-cd46-45de-a0fe-962080b2a80e.metadata.json
		at com.netease.arctic.io.ArcticHadoopFileIO.lambda$deleteFile$2(ArcticHadoopFileIO.java:76) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.table.TableMetaStore.lambda$doAs$0(TableMetaStore.java:315) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_152]
		at javax.security.auth.Subject.doAs(Subject.java:360) ~[?:1.8.0_152]
		at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1734) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:313) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.io.ArcticHadoopFileIO.deleteFile(ArcticHadoopFileIO.java:67) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.hadoop.HadoopTableOperations.tryDelete(HadoopTableOperations.java:386) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.hadoop.HadoopTableOperations.renameToFinal(HadoopTableOperations.java:368) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.hadoop.HadoopTableOperations.commit(HadoopTableOperations.java:158) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.op.ArcticHadoopTableOperations.lambda$commit$1(ArcticHadoopTableOperations.java:45) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.table.TableMetaStore.lambda$doAs$0(TableMetaStore.java:315) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_152]
		at javax.security.auth.Subject.doAs(Subject.java:360) ~[?:1.8.0_152]
		at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1734) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:313) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.io.ArcticHadoopFileIO.doAs(ArcticHadoopFileIO.java:184) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.op.ArcticHadoopTableOperations.commit(ArcticHadoopTableOperations.java:44) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.BaseTransaction.lambda$commitSimpleTransaction$5(BaseTransaction.java:356) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.util.Tasks$Builder.runTaskWithRetry(Tasks.java:404) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.util.Tasks$Builder.runSingleThreaded(Tasks.java:214) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:198) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:190) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.BaseTransaction.commitSimpleTransaction(BaseTransaction.java:348) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.BaseTransaction.commitTransaction(BaseTransaction.java:238) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.op.ArcticUpdate.commit(ArcticUpdate.java:137) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.flink.sink.IcebergFilesCommitter.commitOperation(IcebergFilesCommitter.java:312) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.flink.sink.IcebergFilesCommitter.commitDeltaTxn(IcebergFilesCommitter.java:276) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.flink.sink.IcebergFilesCommitter.commitUpToCheckpoint(IcebergFilesCommitter.java:218) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.flink.sink.IcebergFilesCommitter.notifyCheckpointComplete(IcebergFilesCommitter.java:188) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at sun.reflect.GeneratedMethodAccessor54.invoke(Unknown Source) ~[?:?]
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_152]
		at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_152]
		at com.netease.arctic.flink.interceptor.KerberosInvocationHandler.lambda$invoke$0(KerberosInvocationHandler.java:58) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.table.TableMetaStore.lambda$doAs$0(TableMetaStore.java:315) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_152]
		at javax.security.auth.Subject.doAs(Subject.java:360) ~[?:1.8.0_152]
		at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1734) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:313) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.io.ArcticHadoopFileIO.doAs(ArcticHadoopFileIO.java:184) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.flink.interceptor.KerberosInvocationHandler.invoke(KerberosInvocationHandler.java:55) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.sun.proxy.$Proxy38.notifyCheckpointComplete(Unknown Source) ~[?:?]
		at sun.reflect.GeneratedMethodAccessor54.invoke(Unknown Source) ~[?:?]
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_152]
		at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_152]
		at com.netease.arctic.flink.interceptor.KerberosInvocationHandler.lambda$invoke$0(KerberosInvocationHandler.java:58) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.table.TableMetaStore.lambda$doAs$0(TableMetaStore.java:315) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_152]
		at javax.security.auth.Subject.doAs(Subject.java:360) ~[?:1.8.0_152]
		at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1734) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:313) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.io.ArcticHadoopFileIO.doAs(ArcticHadoopFileIO.java:184) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.flink.interceptor.KerberosInvocationHandler.invoke(KerberosInvocationHandler.java:55) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.sun.proxy.$Proxy38.notifyCheckpointComplete(Unknown Source) ~[?:?]
		at org.apache.flink.streaming.runtime.tasks.StreamOperatorWrapper.notifyCheckpointComplete(StreamOperatorWrapper.java:99) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.streaming.runtime.tasks.SubtaskCheckpointCoordinatorImpl.notifyCheckpointComplete(SubtaskCheckpointCoordinatorImpl.java:319) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.streaming.runtime.tasks.StreamTask.notifyCheckpointComplete(StreamTask.java:1089) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.streaming.runtime.tasks.StreamTask.lambda$notifyCheckpointCompleteAsync$11(StreamTask.java:1054) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.streaming.runtime.tasks.StreamTask.lambda$notifyCheckpointOperation$13(StreamTask.java:1077) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.streaming.runtime.tasks.StreamTaskActionExecutor$1.runThrowing(StreamTaskActionExecutor.java:50) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.streaming.runtime.tasks.mailbox.Mail.run(Mail.java:90) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.processMail(MailboxProcessor.java:317) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.runMailboxLoop(MailboxProcessor.java:189) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.streaming.runtime.tasks.StreamTask.runMailboxLoop(StreamTask.java:619) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.streaming.runtime.tasks.StreamTask.invoke(StreamTask.java:583) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.runtime.taskmanager.Task.doRun(Task.java:758) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.runtime.taskmanager.Task.run(Task.java:573) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at java.lang.Thread.run(Thread.java:748) [?:1.8.0_152]
	Caused by: java.io.IOException: java.lang.InterruptedException
		at org.apache.hadoop.ipc.Client.call(Client.java:1461) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.ipc.Client.call(Client.java:1413) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:229) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at com.sun.proxy.$Proxy27.delete(Unknown Source) ~[?:?]
		at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.delete(ClientNamenodeProtocolTranslatorPB.java:545) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at sun.reflect.GeneratedMethodAccessor44.invoke(Unknown Source) ~[?:?]
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_152]
		at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_152]
		at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:191) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:102) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at com.sun.proxy.$Proxy28.delete(Unknown Source) ~[?:?]
		at org.apache.hadoop.hdfs.DFSClient.delete(DFSClient.java:2053) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.hdfs.DistributedFileSystem$14.doCall(DistributedFileSystem.java:707) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.hdfs.DistributedFileSystem$14.doCall(DistributedFileSystem.java:703) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.fs.FileSystemLinkResolver.resolve(FileSystemLinkResolver.java:81) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.hdfs.DistributedFileSystem.delete(DistributedFileSystem.java:714) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at com.netease.arctic.io.ArcticHadoopFileIO.lambda$deleteFile$2(ArcticHadoopFileIO.java:74) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		... 67 more
	Caused by: java.lang.InterruptedException
		at java.util.concurrent.FutureTask.awaitDone(FutureTask.java:404) ~[?:1.8.0_152]
		at java.util.concurrent.FutureTask.get(FutureTask.java:191) ~[?:1.8.0_152]
		at org.apache.hadoop.ipc.Client$Connection.sendRpcRequest(Client.java:1060) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.ipc.Client.call(Client.java:1455) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.ipc.Client.call(Client.java:1413) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:229) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at com.sun.proxy.$Proxy27.delete(Unknown Source) ~[?:?]
		at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.delete(ClientNamenodeProtocolTranslatorPB.java:545) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at sun.reflect.GeneratedMethodAccessor44.invoke(Unknown Source) ~[?:?]
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_152]
		at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_152]
		at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:191) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:102) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at com.sun.proxy.$Proxy28.delete(Unknown Source) ~[?:?]
		at org.apache.hadoop.hdfs.DFSClient.delete(DFSClient.java:2053) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.hdfs.DistributedFileSystem$14.doCall(DistributedFileSystem.java:707) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.hdfs.DistributedFileSystem$14.doCall(DistributedFileSystem.java:703) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.fs.FileSystemLinkResolver.resolve(FileSystemLinkResolver.java:81) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.hdfs.DistributedFileSystem.delete(DistributedFileSystem.java:714) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at com.netease.arctic.io.ArcticHadoopFileIO.lambda$deleteFile$2(ArcticHadoopFileIO.java:74) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		... 67 more
Caused by: java.io.InterruptedIOException: Call interrupted
	at org.apache.hadoop.ipc.Client.call(Client.java:1470) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.ipc.Client.call(Client.java:1413) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:229) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at com.sun.proxy.$Proxy27.rename(Unknown Source) ~[?:?]
	at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.rename(ClientNamenodeProtocolTranslatorPB.java:487) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at sun.reflect.GeneratedMethodAccessor61.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_152]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_152]
	at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:191) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:102) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at com.sun.proxy.$Proxy28.rename(Unknown Source) ~[?:?]
	at org.apache.hadoop.hdfs.DFSClient.rename(DFSClient.java:1960) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.hdfs.DistributedFileSystem.rename(DistributedFileSystem.java:626) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at com.netease.arctic.shaded.org.apache.iceberg.hadoop.HadoopTableOperations.renameToFinal(HadoopTableOperations.java:356) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	... 59 more
2022-11-25 23:00:00,977 WARN  com.netease.arctic.shaded.org.apache.iceberg.util.Tasks      [] - Retrying task after failure: Failed to commit changes using rename: hdfs://easyops-cluster/user/arctic/hive_db/arctic.db/avro_lost_test/change/metadata/v801.metadata.json
com.netease.arctic.shaded.org.apache.iceberg.exceptions.CommitFailedException: Failed to commit changes using rename: hdfs://easyops-cluster/user/arctic/hive_db/arctic.db/avro_lost_test/change/metadata/v801.metadata.json
	at com.netease.arctic.shaded.org.apache.iceberg.hadoop.HadoopTableOperations.renameToFinal(HadoopTableOperations.java:366) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.hadoop.HadoopTableOperations.commit(HadoopTableOperations.java:158) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.op.ArcticHadoopTableOperations.lambda$commit$1(ArcticHadoopTableOperations.java:45) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.table.TableMetaStore.lambda$doAs$0(TableMetaStore.java:315) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_152]
	at javax.security.auth.Subject.doAs(Subject.java:360) ~[?:1.8.0_152]
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1734) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:313) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.io.ArcticHadoopFileIO.doAs(ArcticHadoopFileIO.java:184) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.op.ArcticHadoopTableOperations.commit(ArcticHadoopTableOperations.java:44) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.BaseTransaction.lambda$commitSimpleTransaction$5(BaseTransaction.java:356) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.util.Tasks$Builder.runTaskWithRetry(Tasks.java:404) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.util.Tasks$Builder.runSingleThreaded(Tasks.java:214) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:198) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:190) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.BaseTransaction.commitSimpleTransaction(BaseTransaction.java:348) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.BaseTransaction.commitTransaction(BaseTransaction.java:238) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.op.ArcticUpdate.commit(ArcticUpdate.java:137) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.flink.sink.IcebergFilesCommitter.commitOperation(IcebergFilesCommitter.java:312) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.flink.sink.IcebergFilesCommitter.commitDeltaTxn(IcebergFilesCommitter.java:276) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.flink.sink.IcebergFilesCommitter.commitUpToCheckpoint(IcebergFilesCommitter.java:218) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.shaded.org.apache.iceberg.flink.sink.IcebergFilesCommitter.notifyCheckpointComplete(IcebergFilesCommitter.java:188) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at sun.reflect.GeneratedMethodAccessor54.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_152]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_152]
	at com.netease.arctic.flink.interceptor.KerberosInvocationHandler.lambda$invoke$0(KerberosInvocationHandler.java:58) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.table.TableMetaStore.lambda$doAs$0(TableMetaStore.java:315) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_152]
	at javax.security.auth.Subject.doAs(Subject.java:360) ~[?:1.8.0_152]
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1734) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:313) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.io.ArcticHadoopFileIO.doAs(ArcticHadoopFileIO.java:184) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.flink.interceptor.KerberosInvocationHandler.invoke(KerberosInvocationHandler.java:55) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.sun.proxy.$Proxy38.notifyCheckpointComplete(Unknown Source) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor54.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_152]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_152]
	at com.netease.arctic.flink.interceptor.KerberosInvocationHandler.lambda$invoke$0(KerberosInvocationHandler.java:58) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.table.TableMetaStore.lambda$doAs$0(TableMetaStore.java:315) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_152]
	at javax.security.auth.Subject.doAs(Subject.java:360) ~[?:1.8.0_152]
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1734) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:313) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.io.ArcticHadoopFileIO.doAs(ArcticHadoopFileIO.java:184) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.netease.arctic.flink.interceptor.KerberosInvocationHandler.invoke(KerberosInvocationHandler.java:55) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at com.sun.proxy.$Proxy38.notifyCheckpointComplete(Unknown Source) ~[?:?]
	at org.apache.flink.streaming.runtime.tasks.StreamOperatorWrapper.notifyCheckpointComplete(StreamOperatorWrapper.java:99) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.streaming.runtime.tasks.SubtaskCheckpointCoordinatorImpl.notifyCheckpointComplete(SubtaskCheckpointCoordinatorImpl.java:319) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.notifyCheckpointComplete(StreamTask.java:1089) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.lambda$notifyCheckpointCompleteAsync$11(StreamTask.java:1054) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.lambda$notifyCheckpointOperation$13(StreamTask.java:1077) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.streaming.runtime.tasks.StreamTaskActionExecutor$1.runThrowing(StreamTaskActionExecutor.java:50) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.streaming.runtime.tasks.mailbox.Mail.run(Mail.java:90) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.processMail(MailboxProcessor.java:317) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.runMailboxLoop(MailboxProcessor.java:189) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.runMailboxLoop(StreamTask.java:619) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.invoke(StreamTask.java:583) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.runtime.taskmanager.Task.doRun(Task.java:758) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at org.apache.flink.runtime.taskmanager.Task.run(Task.java:573) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_152]
	Suppressed: java.io.UncheckedIOException: Failed to delete file: hdfs://easyops-cluster/user/arctic/hive_db/arctic.db/avro_lost_test/change/metadata/13322137-cd46-45de-a0fe-962080b2a80e.metadata.json
		at com.netease.arctic.io.ArcticHadoopFileIO.lambda$deleteFile$2(ArcticHadoopFileIO.java:76) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.table.TableMetaStore.lambda$doAs$0(TableMetaStore.java:315) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_152]
		at javax.security.auth.Subject.doAs(Subject.java:360) ~[?:1.8.0_152]
		at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1734) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:313) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.io.ArcticHadoopFileIO.deleteFile(ArcticHadoopFileIO.java:67) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.hadoop.HadoopTableOperations.tryDelete(HadoopTableOperations.java:386) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.hadoop.HadoopTableOperations.renameToFinal(HadoopTableOperations.java:368) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.hadoop.HadoopTableOperations.commit(HadoopTableOperations.java:158) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.op.ArcticHadoopTableOperations.lambda$commit$1(ArcticHadoopTableOperations.java:45) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.table.TableMetaStore.lambda$doAs$0(TableMetaStore.java:315) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_152]
		at javax.security.auth.Subject.doAs(Subject.java:360) ~[?:1.8.0_152]
		at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1734) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:313) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.io.ArcticHadoopFileIO.doAs(ArcticHadoopFileIO.java:184) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.op.ArcticHadoopTableOperations.commit(ArcticHadoopTableOperations.java:44) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.BaseTransaction.lambda$commitSimpleTransaction$5(BaseTransaction.java:356) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.util.Tasks$Builder.runTaskWithRetry(Tasks.java:404) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.util.Tasks$Builder.runSingleThreaded(Tasks.java:214) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:198) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:190) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.BaseTransaction.commitSimpleTransaction(BaseTransaction.java:348) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.BaseTransaction.commitTransaction(BaseTransaction.java:238) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.op.ArcticUpdate.commit(ArcticUpdate.java:137) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.flink.sink.IcebergFilesCommitter.commitOperation(IcebergFilesCommitter.java:312) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.flink.sink.IcebergFilesCommitter.commitDeltaTxn(IcebergFilesCommitter.java:276) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.flink.sink.IcebergFilesCommitter.commitUpToCheckpoint(IcebergFilesCommitter.java:218) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.shaded.org.apache.iceberg.flink.sink.IcebergFilesCommitter.notifyCheckpointComplete(IcebergFilesCommitter.java:188) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at sun.reflect.GeneratedMethodAccessor54.invoke(Unknown Source) ~[?:?]
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_152]
		at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_152]
		at com.netease.arctic.flink.interceptor.KerberosInvocationHandler.lambda$invoke$0(KerberosInvocationHandler.java:58) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.table.TableMetaStore.lambda$doAs$0(TableMetaStore.java:315) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_152]
		at javax.security.auth.Subject.doAs(Subject.java:360) ~[?:1.8.0_152]
		at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1734) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:313) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.io.ArcticHadoopFileIO.doAs(ArcticHadoopFileIO.java:184) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.flink.interceptor.KerberosInvocationHandler.invoke(KerberosInvocationHandler.java:55) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.sun.proxy.$Proxy38.notifyCheckpointComplete(Unknown Source) ~[?:?]
		at sun.reflect.GeneratedMethodAccessor54.invoke(Unknown Source) ~[?:?]
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_152]
		at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_152]
		at com.netease.arctic.flink.interceptor.KerberosInvocationHandler.lambda$invoke$0(KerberosInvocationHandler.java:58) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.table.TableMetaStore.lambda$doAs$0(TableMetaStore.java:315) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_152]
		at javax.security.auth.Subject.doAs(Subject.java:360) ~[?:1.8.0_152]
		at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1734) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:313) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.io.ArcticHadoopFileIO.doAs(ArcticHadoopFileIO.java:184) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.netease.arctic.flink.interceptor.KerberosInvocationHandler.invoke(KerberosInvocationHandler.java:55) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at com.sun.proxy.$Proxy38.notifyCheckpointComplete(Unknown Source) ~[?:?]
		at org.apache.flink.streaming.runtime.tasks.StreamOperatorWrapper.notifyCheckpointComplete(StreamOperatorWrapper.java:99) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.streaming.runtime.tasks.SubtaskCheckpointCoordinatorImpl.notifyCheckpointComplete(SubtaskCheckpointCoordinatorImpl.java:319) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.streaming.runtime.tasks.StreamTask.notifyCheckpointComplete(StreamTask.java:1089) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.streaming.runtime.tasks.StreamTask.lambda$notifyCheckpointCompleteAsync$11(StreamTask.java:1054) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.streaming.runtime.tasks.StreamTask.lambda$notifyCheckpointOperation$13(StreamTask.java:1077) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.streaming.runtime.tasks.StreamTaskActionExecutor$1.runThrowing(StreamTaskActionExecutor.java:50) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.streaming.runtime.tasks.mailbox.Mail.run(Mail.java:90) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.processMail(MailboxProcessor.java:317) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.runMailboxLoop(MailboxProcessor.java:189) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.streaming.runtime.tasks.StreamTask.runMailboxLoop(StreamTask.java:619) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.streaming.runtime.tasks.StreamTask.invoke(StreamTask.java:583) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.runtime.taskmanager.Task.doRun(Task.java:758) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at org.apache.flink.runtime.taskmanager.Task.run(Task.java:573) [plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		at java.lang.Thread.run(Thread.java:748) [?:1.8.0_152]
	Caused by: java.io.IOException: java.lang.InterruptedException
		at org.apache.hadoop.ipc.Client.call(Client.java:1461) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.ipc.Client.call(Client.java:1413) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:229) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at com.sun.proxy.$Proxy27.delete(Unknown Source) ~[?:?]
		at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.delete(ClientNamenodeProtocolTranslatorPB.java:545) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at sun.reflect.GeneratedMethodAccessor44.invoke(Unknown Source) ~[?:?]
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_152]
		at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_152]
		at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:191) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:102) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at com.sun.proxy.$Proxy28.delete(Unknown Source) ~[?:?]
		at org.apache.hadoop.hdfs.DFSClient.delete(DFSClient.java:2053) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.hdfs.DistributedFileSystem$14.doCall(DistributedFileSystem.java:707) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.hdfs.DistributedFileSystem$14.doCall(DistributedFileSystem.java:703) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.fs.FileSystemLinkResolver.resolve(FileSystemLinkResolver.java:81) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.hdfs.DistributedFileSystem.delete(DistributedFileSystem.java:714) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at com.netease.arctic.io.ArcticHadoopFileIO.lambda$deleteFile$2(ArcticHadoopFileIO.java:74) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		... 67 more
	Caused by: java.lang.InterruptedException
		at java.util.concurrent.FutureTask.awaitDone(FutureTask.java:404) ~[?:1.8.0_152]
		at java.util.concurrent.FutureTask.get(FutureTask.java:191) ~[?:1.8.0_152]
		at org.apache.hadoop.ipc.Client$Connection.sendRpcRequest(Client.java:1060) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.ipc.Client.call(Client.java:1455) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.ipc.Client.call(Client.java:1413) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:229) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at com.sun.proxy.$Proxy27.delete(Unknown Source) ~[?:?]
		at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.delete(ClientNamenodeProtocolTranslatorPB.java:545) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at sun.reflect.GeneratedMethodAccessor44.invoke(Unknown Source) ~[?:?]
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_152]
		at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_152]
		at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:191) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:102) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at com.sun.proxy.$Proxy28.delete(Unknown Source) ~[?:?]
		at org.apache.hadoop.hdfs.DFSClient.delete(DFSClient.java:2053) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.hdfs.DistributedFileSystem$14.doCall(DistributedFileSystem.java:707) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.hdfs.DistributedFileSystem$14.doCall(DistributedFileSystem.java:703) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.fs.FileSystemLinkResolver.resolve(FileSystemLinkResolver.java:81) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at org.apache.hadoop.hdfs.DistributedFileSystem.delete(DistributedFileSystem.java:714) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
		at com.netease.arctic.io.ArcticHadoopFileIO.lambda$deleteFile$2(ArcticHadoopFileIO.java:74) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
		... 67 more
Caused by: java.io.InterruptedIOException: Call interrupted
	at org.apache.hadoop.ipc.Client.call(Client.java:1470) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.ipc.Client.call(Client.java:1413) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:229) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at com.sun.proxy.$Proxy27.rename(Unknown Source) ~[?:?]
	at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.rename(ClientNamenodeProtocolTranslatorPB.java:487) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at sun.reflect.GeneratedMethodAccessor61.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_152]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_152]
	at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:191) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:102) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at com.sun.proxy.$Proxy28.rename(Unknown Source) ~[?:?]
	at org.apache.hadoop.hdfs.DFSClient.rename(DFSClient.java:1960) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at org.apache.hadoop.hdfs.DistributedFileSystem.rename(DistributedFileSystem.java:626) ~[flink-shaded-hadoop-2-uber-2.7.5-10.0.jar:2.7.5-10.0]
	at com.netease.arctic.shaded.org.apache.iceberg.hadoop.HadoopTableOperations.renameToFinal(HadoopTableOperations.java:356) ~[plugin_ne-flink-1.12.4-1.1.7_scala2.12_hive2.1.1-3.9.4-1.4.4.1-snap-test.jar:?]
	... 59 more

--- a/core/src/main/java/com/netease/arctic/op/ArcticHadoopTableOperations.java
+++ b/core/src/main/java/com/netease/arctic/op/ArcticHadoopTableOperations.java
@@ -59,8 +59,6 @@ public class ArcticHadoopTableOperations extends HadoopTableOperations {
           // Do to wrap the direct CommitFailedException, we should retry committing.
           throw e;
         }
-      } catch (RuntimeException e) {
-        throw new CommitStateUnknownException(e);
       }
       return null;
     });

--- a/core/src/main/java/com/netease/arctic/op/ArcticHadoopTableOperations.java
+++ b/core/src/main/java/com/netease/arctic/op/ArcticHadoopTableOperations.java
@@ -22,6 +22,7 @@ import com.netease.arctic.io.ArcticFileIO;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.fs.Path;
 import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.exceptions.CommitStateUnknownException;
 import org.apache.iceberg.hadoop.HadoopTableOperations;
 import org.apache.iceberg.util.LockManagers;
 
@@ -42,7 +43,16 @@ public class ArcticHadoopTableOperations extends HadoopTableOperations {
   @Override
   public void commit(TableMetadata base, TableMetadata metadata) {
     arcticFileIO.doAs(() -> {
-      super.commit(base, metadata);
+      try {
+        super.commit(base, metadata);
+      } catch (RuntimeException e) {
+        // HadoopTableOperations#commit will throw CommitFailedException even though rename metadata file successfully
+        // in hdfs, it may be not safe. So transform all RuntimeException to CommitStateUnknownException to avoid
+        // delete the committed metadata and manifest files.
+        //
+        // But this change may invalid the retry action for all commit operation.
+        throw new CommitStateUnknownException(e);
+      }
       return null;
     });
   }

--- a/core/src/main/java/com/netease/arctic/op/ArcticHadoopTableOperations.java
+++ b/core/src/main/java/com/netease/arctic/op/ArcticHadoopTableOperations.java
@@ -22,6 +22,7 @@ import com.netease.arctic.io.ArcticFileIO;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.fs.Path;
 import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.exceptions.CommitFailedException;
 import org.apache.iceberg.exceptions.CommitStateUnknownException;
 import org.apache.iceberg.hadoop.HadoopTableOperations;
 import org.apache.iceberg.util.LockManagers;
@@ -45,12 +46,20 @@ public class ArcticHadoopTableOperations extends HadoopTableOperations {
     arcticFileIO.doAs(() -> {
       try {
         super.commit(base, metadata);
-      } catch (RuntimeException e) {
+
         // HadoopTableOperations#commit will throw CommitFailedException even though rename metadata file successfully
         // in hdfs, it may be not safe. So transform all RuntimeException to CommitStateUnknownException to avoid
         // delete the committed metadata and manifest files.
         //
-        // But this change may invalid the retry action for all commit operation.
+        // But this change may invalid the retry action for some commit operation.
+      } catch (CommitFailedException e) {
+        if (e.getCause() != null) {
+          throw new CommitStateUnknownException(e);
+        } else {
+          // Do to wrap the direct CommitFailedException, we should retry committing.
+          throw e;
+        }
+      } catch (RuntimeException e) {
         throw new CommitStateUnknownException(e);
       }
       return null;

